src/modules/gnd_pos_control/GroundRoverPositionControl.cpp

```diff
@@ -447,22 +444,22 @@ GroundRoverPositionControl::task_main()
 
 				/* lazily publish navigation capabilities */
 				if ((hrt_elapsed_time(&_gnd_pos_ctrl_status.timestamp) > 1000000)
-				    || (fabsf(turn_distance - _gnd_pos_ctrl_status.turn_distance) > FLT_EPSILON
+				    || (fabsf(turn_distance - _gnd_pos_ctrl_status.acceptance_radius) > FLT_EPSILON
 					&& turn_distance > 0)) {
 
 					/* set new turn distance */
-					_gnd_pos_ctrl_status.turn_distance = turn_distance;
-
-					_gnd_pos_ctrl_status.nav_roll = _gnd_control.get_roll_setpoint();
-					_gnd_pos_ctrl_status.nav_pitch = 0.0f;
-					_gnd_pos_ctrl_status.nav_bearing = _gnd_control.nav_bearing();
-
-					_gnd_pos_ctrl_status.target_bearing = _gnd_control.target_bearing();
-					_gnd_pos_ctrl_status.xtrack_error = _gnd_control.crosstrack_error();
-
-					matrix::Vector2f curr_wp((float)_pos_sp_triplet.current.lat, (float)_pos_sp_triplet.current.lon);
-					_gnd_pos_ctrl_status.wp_dist = get_distance_to_next_waypoint((double)current_position(0), (double)current_position(1),
-								       (double)curr_wp(0), (double)curr_wp(1));
+					_gnd_pos_ctrl_status.acceptance_radius = turn_distance;
+
+//					_gnd_pos_ctrl_status.nav_roll = _gnd_control.get_roll_setpoint();
```

Comment:
Why is this code here just commented out?

Reply:
@dagar how can we bring this in? Could we use this new message only for avoidance and then in a new PR use it for VTOL as well?

Reply:
@mrivi You are not filling the payload of the message but still publishing it below - is this intended?

Reply:
This was from the original PR where fw_pos_ctrl_status was being generalized. Some of these fields are straight copies of what also goes out in vehicle_attitude_setpoint, so I was going to drop them.
https://github.com/PX4/Firmware/pull/10037

Let's finish https://github.com/PX4/Firmware/pull/10037 and rebase.

Reply:
@dagar I have rebased.